### PR TITLE
Dashboard page with all the feeds

### PR DESCRIPTION
**Pull request by goodnessaig1**

### CODE REVIEW

File: src/Components/Dashboard/MainPage.js [hunk 1]
```diff
@@ -1,14 +1,232 @@
...
+import moment from 'moment';
+import commentIcon from '../Assets/comment.png'
 
-const MainPage = () => {
 
 
+const MainPage = ({feeds,user}) => {
+    
+    const dispatch = useDispatch()
+    const getBackgroundColor = (value) => {
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: this should be a utility function in a utils file

File: src/Components/Dashboard/MainPage.js [hunk 2]
```diff
@@ -1,14 +1,232 @@
...
+          color = ' #e0c375';
+      } else if (value >= 160 && value < 170) {
+          color = ' #d83030';
+      }
+      return color;
+  };
+    useEffect(() => {
+        dispatch(getFeedDetails())
+    },[])
+    console.log(feeds)
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: remove console

File: src/Components/Dashboard/MainPage.js [hunk 3]
```diff
@@ -1,14 +1,232 @@
...
   return (
     <PageLayout>
-        <div>Dashboard and other pages</div>
+      <div className='dashboard_container'>
+        <div className='dash_board_top'>
+            <div className='dashboard_top'>
+              {
+                user && user.profile !== null ?(
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: you can simplify this by just doing `user?.profile ? `

File: src/Components/Dashboard/MainPage.js [hunk 4]
```diff
@@ -1,14 +1,232 @@
...
+                   <img src={user.profile} alt='' className='profile__image' />
+                ):
+                   <img src={Unavailiabe} className='profile__image' alt='' />
+              }
+                <input type='text' placeholder='Share your thoughts...' className='share_input'/>
+            </div>
+        </div>
+    
+                              
+                                      {/*========    MAIN FEEDS ===========*/}
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: please create a component to render this

File: src/Components/Dashboard/MainPage.js [hunk 5]
```diff
@@ -1,14 +1,232 @@
...
+                  feeds && feeds.map((item, index) =>
+                    <div key={index}>
+                    <div className='feed_content' >
+                      {
+                        item.post.includes("https://") ?(
+                          <div className='feed_content'>
+                            <div  className='feed_top'>
+                              {
+                                item.profile_pix !== null ? (
+                                  <img src={item.profile_pix} className='profile__pix' alt='' />
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: this is repeating itself too much. create a component to render profile pix. it should render the unavalaible when its not there and render when there. then just pass the item.profile_pix as a prop

File: src/Components/Dashboard/MainPage.js [hunk 6]
```diff
@@ -1,14 +1,232 @@
...
+                        </div>
+                            <div className='post_container'>
+                                <div className='gif_post_title'>{item.title}</div>
+                                <img src={item.post}  className='post' alt='' />
+                            </div>
+                            <div className='like_comment_container'>
+                              <div className='like'>
+                                {
+                                  item.isliked === false ? (
+                                    <img className='like_img' src={Like} alt=''/>
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: please name this properly. one is Like and the other Liked. don't get?

File: src/Components/Dashboard/MainPage.js [hunk 7]
```diff
@@ -1,14 +1,232 @@
...
+                                <span>{item.number_of_likes}</span>
+                              </div>
+                              <div className='comment'>
+                                <img src={commentIcon} className='comment_Icon' alt=''/>
+                                <span>{item.number_of_commennt}</span>
+                              </div>
+                            </div>
+                            
+                              {
+                              item.comment !== null ? (
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: why am i seeing just one comment?? what if there are several comments?

**Review by courage173 — CHANGES_REQUESTED** [review]
you have alot of work to do on this. please review the comments. also i want you to modify how you make your api calls on redux, you should also dispatch when a request is made, that way you can show a loading spinner. what i mean is this.
1. make request to get all feeds - i dispatch a request action to set isLoading = true
2. successful you then dispatch a successful action like you currently do
3. failed same with what you currently do. the first one is missing.

Please go through each of my comments and know when to break codes into components instead of one file being so big. and also your code looks dirty and thats because your eslint and prettier are not formatting your code properly. please fix this in your next push too.

check this for reference for the redux and copy the .eslintrc.json file and install the necessary eslint and prettier packages there.
[https://github.com/courage173/boum-chat-app](https://github.com/courage173/boum-chat-app)

File: src/Auth/Actions/feedActions.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+export const getAllFeedsFailure = error => {
+    return {
+        type: GET_ALL_FEEDS_FAILURE,
+        payload: error,
+    };
+};
+
+export function getFeedDetails(offset, setIsLoading) {
+    return dispatch => {
+        setIsLoading(true);
```
Comment (courage173) [inline src/Auth/Actions/feedActions.js]: i did not say setIsLoading. please do this with redux

File: src/Components/Dashboard/MainPage.js [hunk 9]
```diff
@@ -1,14 +1,71 @@
...
+    const feedsLength = feeds?.length;
+    const dispatch = useDispatch();
+    const [isLoading, setIsLoading] = useState(false);
+    useEffect(() => {
+        window.scrollTo({
+            top: 0,
+            left: 0,
+            behavior: 'smooth',
+        });
+        dispatch(getFeedDetails(offSet, setIsLoading));
```
Comment (courage173) [inline src/Components/Dashboard/MainPage.js]: why is this still here? get the loading state from redux store now

**Review by courage173 — CHANGES_REQUESTED** [review]
also the requesting change for others

File: src/Components/Dashboard/MainPage.js (hunk 9 above)
Comment (goodnessaig1) [reply src/Components/Dashboard/MainPage.js]: Ok sir. I have now been able to get the loading state from redux. And i have implemented it in my code

**Review by courage173 — CHANGES_REQUESTED** [review]
also the requesting change for others

File: src/Auth/Reducer/userReducer.js [hunk 10]
```diff
@@ -1,53 +1,210 @@
...
+                    error: null,
+                    success: true,
+                },
+                user: action.payload,
+            });
+        case types.LOGIN_USER_FAILURE:
+            return Object.assign({}, state, {
+                LoginUser: {
+                    requesting: false,
+                    error: null,
```
Comment (courage173) [inline src/Auth/Reducer/userReducer.js:90]: why is the error null for a failure

File: src/Auth/Reducer/userReducer.js [hunk 11]
```diff
@@ -1,53 +1,210 @@
...
+                    success: true,
+                },
+                userData: action.payload,
+            });
+        case types.GET_USER_DETAILS_FAILURE:
+            return Object.assign({}, state, {
+                getUserDetails: {
+                    requesting: false,
+                    error: null,
```
Comment (courage173) [inline src/Auth/Reducer/userReducer.js]: same

File: src/Components/Profile/UserCoverModal.js [hunk 12]
```diff
@@ -1,125 +1,141 @@
...
+                setPreview(reader.result);
+            };
         }
     };
 
-    const handleCancle = (e)=>{
-        if (e.target){
-            setPreview([])
+    const handleClick = e => {
+        if (e.target.classList.contains('dismiss')) {
```
Comment (courage173) [inline src/Components/Profile/UserCoverModal.js]: why do you do this check please?

Comment (courage173) [reply src/Components/Profile/UserCoverModal.js]: why are you relying on this ?? e.target.classList.contains('dismiss')

**Review by courage173 — CHANGES_REQUESTED** [review]

**Review by courage173 — APPROVED** [review]

**Pull request merged**